daemon/container_operations.go

```diff
@@ -719,6 +719,13 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 		return err
 	}
 
+	if !container.Managed {
```

Comment:
@mavenugo yes, `!isClusterEligible()` will make it a noop for non-swarm networks. The same check was relied on earlier (before the addition of `EnableService()` API) when `addToCluster` was getting called unconditionally.